test/core/iomgr/tcp_client_posix_test.cc

```diff
@@ -190,6 +190,45 @@ void test_fails(void) {
   gpr_log(GPR_ERROR, "---- finished test_fails() ----");
 }
 
+void test_connect_cancellation_succeeds(void) {
```

Comment:
How about also testing the case where the cancellation fails because the connection attempt is already complete?

Reply:
Yes thats a good idea. Added this in lines 198 and 145